src/prefect/engine.py

```diff
@@ -117,11 +118,16 @@ def my_flow():
             )
             raise
 
-        # run the flow
-        if flow.isasync:
-            run_coro_as_sync(run_flow(flow, flow_run=flow_run, error_logger=run_logger))
-        else:
-            run_flow(flow, flow_run=flow_run, error_logger=run_logger)
+        try:
```

Comment:
We might want to move this `try` block to include the `load_flow` call too since we might miss logs when load flow fails. Either than or call `QueueService.drain_all` when we encounter an exception there.

Reply:
Oh good point, I'll do that

Reply:
I went with adding another explicit drain in the `except` here because I didn't want to drain twice under normal conditions